infra/modules/providers/azure/app-gateway/main.tf

```diff
@@ -0,0 +1,69 @@
+data "azurerm_resource_group" "appgateway" {
+  name      = "${var.resource_group_name}"
+}
+
+data "azurerm_virtual_network" "appgateway" {
+    name                = "${var.virtual_network_name}"
+    resource_group_name = "${data.azurerm_resource_group.appgateway.name}"
+}
+data "azurerm_subnet" "appgateway" {
+    name                    = "${var.subnet_name}"
+    resource_group_name     = "${data.azurerm_resource_group.appgateway.name}"
+    virtual_network_name    = "${data.azurerm_virtual_network.appgateway.name}"
+}
+
+resource "azurerm_application_gateway" "appgateway" {
+  name                = "${var.appgateway_name}"
+  resource_group_name = "${data.azurerm_resource_group.appgateway.name}"
+  location            = "${data.azurerm_resource_group.appgateway.location}"
+  tags                = "${var.resource_tags}"
+
+  sku {
+    name     = "${var.appgateway_sku_name}"
+    tier     = "${var.appgateway_tier}"
+    capacity = "${var.appgateway_capacity}"
+  }
+
+  gateway_ip_configuration {
+    name      = "${var.appgateway_ipconfig_name}"
+    subnet_id = "${data.azurerm_subnet.appgateway.id}"
+  }
+
+  frontend_port {
+    name = "${var.appgateway_frontend_port_name}"
+    port = "${var.frontend_http_port}"
+  }
+
+  frontend_ip_configuration {
+    name                  = "${var.appgateway_frontend_ip_configuration_name}"
+    subnet_id             = "${data.azurerm_subnet.appgateway.id}"
+    private_ip_address    = "${var.frontend_ip_config_private_ip_address}"
+    public_ip_address_id  = "${var.frontend_ip_config_public_ip_address_id}"
```

Comment:
Can you please confirm that you tested this module for instanced when `frontend_ip_config_public_ip_address_id` is null and not null?

Reply:
I made some changes and tested for frontend IP configuration to work with and without vnets. Now subnet ID is an optional variable as compared to required variable. It works with frontend_ip_config_public_ip_address_id as null and not null.